src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -314,11 +314,17 @@ public async Task<Document> ApplyCodeFixesForSpecificDiagnosticIdAsync(Document
 
             var extensionManager = document.Project.Solution.Workspace.Services.GetService<IExtensionManager>();
 
-            // Run each CodeFixProvider to gather individual CodeFixes for reported diagnostics
-            // Ensure that each diagnostic only has a unique registered code action for any given equivalance key.
+            // Run each CodeFixProvider to gather individual CodeFixes for reported diagnostics.
+            // Ensure that no diagnostic has registered code actions from different code fix providers with same equivalance key.
             // This prevents duplicate registered code actions from NuGet and VSIX code fix providers.
             // See https://github.com/dotnet/roslyn/issues/18818 for details.
             var uniqueDiagosticToEquivalenceKeysMap = new Dictionary<Diagnostic, PooledHashSet<string?>>();
+
+            // NOTE: For backward compatibility, we allow multiple registered code actions from the same code fix provider
```

Comment:
💭 It doesn't seem necessary to say it's for backward compatibility. It's just a comment on the design choice which remains relevant.

Reply:
I think that part is relevant - ideally, if we were redesigning this feature, we would disallow single provider to register multiple code actions with same equivalence key as FixAll becomes ambiguous. A back compat note also provides better documentation and guarantee that we will not change the design again.